### PR TITLE
BUG: pivot_table with margins=T raises when results in empty df

### LINKED ISSUE #49240: BUG: pivot_table with margins=True on empty DataFrame results in ValueError
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

pd.DataFrame({'A': [], 'B':[], 'C': []}).pivot_table(values='A', index='B', columns='C', margins=True)
```

### Issue Description

`pivot_table` with `margins=True` on empty DataFrame results in `ValueError`.
This is an inconsistent behaviour, as `pivot_table` with `margins=False` returns empty dataframe.

### Expected Behavior

Should return an empty DataFrame, same as with `margins=False`

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 91111fd99898d9dcaa6bf6bedb662db4108da6e6
python           : 3.10.0.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.17763
machine          : AMD64
processor        : Intel64 Family 6 Model 85 Stepping 7, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_United States.1251

pandas           : 1.5.1
numpy            : 1.21.4
pytz             : 2021.3
dateutil         : 2.8.2
setuptools       : 57.4.0
pip              : 22.3
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : 3.0.3
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 7.30.1
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : 3.5.0
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by debnathshoham**
- [x] closes #49240 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v2.0.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/reshape/pivot.py [hunk 1]
```diff
@@ -411,9 +411,13 @@ def _all_key(key):
                 table_pieces.append(transformed_piece)
                 margin_keys.append(all_key)
 
-        result = concat(table_pieces, axis=cat_axis)
+        try:
```
Comment (phofl) [inline pandas/core/reshape/pivot.py:414]: Table pieces is empty here? Can we just check for
That explicitly instead of try except?

Comment (debnathshoham) [reply pandas/core/reshape/pivot.py]: done.
I have still returned the df table, as that still contains the cols and index

File: doc/source/whatsnew/v2.0.0.rst [hunk 2]
```diff
@@ -441,7 +441,7 @@ Reshaping
 - Bug in :meth:`DataFrame.unstack` and :meth:`Series.unstack` unstacking wrong level of :class:`MultiIndex` when :class:`MultiIndex` has mixed names (:issue:`48763`)
 - Bug in :meth:`DataFrame.pivot` not respecting ``None`` as column name (:issue:`48293`)
 - Bug in :func:`join` when ``left_on`` or ``right_on`` is or includes a :class:`CategoricalIndex` incorrectly raising ``AttributeError`` (:issue:`48464`)
--
+- Bug in :meth:`DataFrame.pivot_table` raising ``ValueError`` with parameter ``margins=True`` when result is an empty DataFrame (:issue:`49240`)
```
Comment (phofl) [inline doc/source/whatsnew/v2.0.0.rst:444]: can you do :class:`DataFrame`?

File: pandas/core/reshape/pivot.py (hunk 1 above)
Comment (phofl) [reply pandas/core/reshape/pivot.py]: Yep this is ok, just the try except was bothering me

File: doc/source/whatsnew/v2.0.0.rst (hunk 2 above)
Comment (debnathshoham) [reply doc/source/whatsnew/v2.0.0.rst]: done

**Review by phofl — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @debnathshoham

**Pull request merged**